pkg/ee/metering/reconcile.go

```diff
@@ -238,8 +237,8 @@ func undeploy(ctx context.Context, client ctrlruntimeclient.Client, namespace st
 }
 
 // cleanupLegacyMeteringResources removes all active parts of the legacy metering installation.
-func cleanupLegacyMeteringResources(ctx context.Context, client ctrlruntimeclient.Client) error {
-	key := types.NamespacedName{Namespace: resources.KubermaticNamespace, Name: meteringToolName}
+func cleanupLegacyMeteringResources(ctx context.Context, client ctrlruntimeclient.Client, namespace string) error {
```

Comment:
We need to keep the hardcoded namespace here, as the legacy components where always deployed to kubermatic namespace. 

Ref: https://github.com/kubermatic/kubermatic/blob/release/v2.20/pkg/ee/metering/reconcile.go#L53

Reply:
Good catch!